features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListener.kt

```diff
@@ -273,13 +273,40 @@ internal class JankStatsActivityLifecycleListener(
     @RequiresApi(Build.VERSION_CODES.N)
     inner class DDFrameMetricsListener : Window.OnFrameMetricsAvailableListener {
 
-        @RequiresApi(Build.VERSION_CODES.S)
+        @RequiresApi(Build.VERSION_CODES.N)
         override fun onFrameMetricsAvailable(
             window: Window,
             frameMetrics: FrameMetrics,
             dropCountSinceLastInvocation: Int
         ) {
-            frameDeadline = frameMetrics.getMetric(FrameMetrics.DEADLINE)
+            delegates.forEach { it.onFrameMetricsData(frameMetricsData.update(frameMetrics)) }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    private fun FrameMetricsData.update(frameMetrics: FrameMetrics) = apply {
+        displayRefreshRate = display?.refreshRate?.toDouble() ?: SIXTY_FPS
+        if (buildSdkVersionProvider.version >= Build.VERSION_CODES.N) {
+            unknownDelayDuration = frameMetrics.getMetric(FrameMetrics.UNKNOWN_DELAY_DURATION)
+            inputHandlingDuration = frameMetrics.getMetric(FrameMetrics.INPUT_HANDLING_DURATION)
+            animationDuration = frameMetrics.getMetric(FrameMetrics.ANIMATION_DURATION)
+            layoutMeasureDuration = frameMetrics.getMetric(FrameMetrics.LAYOUT_MEASURE_DURATION)
+            drawDuration = frameMetrics.getMetric(FrameMetrics.DRAW_DURATION)
+            syncDuration = frameMetrics.getMetric(FrameMetrics.SYNC_DURATION)
+            commandIssueDuration = frameMetrics.getMetric(FrameMetrics.COMMAND_ISSUE_DURATION)
+            swapBuffersDuration = frameMetrics.getMetric(FrameMetrics.SWAP_BUFFERS_DURATION)
+            totalDuration = frameMetrics.getMetric(FrameMetrics.TOTAL_DURATION)
+            firstDrawFrame = frameMetrics.getMetric(FrameMetrics.FIRST_DRAW_FRAME) == TRUE
+        }
+        @SuppressLint("InlinedApi")
+        if (buildSdkVersionProvider.version >= Build.VERSION_CODES.O) {
+            intendedVsyncTimestamp = frameMetrics.getMetric(FrameMetrics.INTENDED_VSYNC_TIMESTAMP)
+            vsyncTimestamp = frameMetrics.getMetric(FrameMetrics.VSYNC_TIMESTAMP)
+        }
+        @SuppressLint("InlinedApi")
```

Comment:
why these annotations are needed? If removed, what is the error?

Reply:
We're using `buildSdkVersionProvider` for OS version resolution. Android Studios's linter is not aware of it and highlighting inappropriate api usage: 
<img width="640" alt="image" src="https://github.com/user-attachments/assets/d7b104bb-e132-4312-8c44-da6c3304d76c" />